src/librustdoc/clean/types.rs

```diff
@@ -1522,7 +1522,7 @@ impl Type {
 
     fn inner_def_id(&self, cache: Option<&Cache>) -> Option<DefId> {
         let t: PrimitiveType = match *self {
-            ResolvedPath { did, .. } => return Some(did),
+            ResolvedPath { ref path, did: _ } => return Some(path.def_id()),
```

Comment:
IMO, *everything* in rustdoc is a good candidate for #91113 😅 

It's very hard to know for sure where this is used because I don't know where `ResolvedPath`s will crop up. My thinking is that the new behavior is "more correct" and could potentially fix bugs.

Reply:
> My thinking is that the new behavior is "more correct" and could potentially fix bugs.

I don't have a lot of confidence in this, unfortunately, a lot of rustdoc has pretty arbitrary behavior that depends on global state elsewhere ... I know this is a pain, but I think we should track down what the actual behavior change is before merging.

Reply:
One thing I've done with past changes is document the stdlib before and after and diff them. Would doing that help increase confidence in the change?

Reply:
Actually, what if I just change it so that `inner_def_id` replicates the old (weird) behavior? Then we do an incremental refactoring where we have more confidence in each of the individual changes.

Reply:
Both of those would give me a lot more confidence :) if the standard library doesn't show any changes we can land this as is, and otherwise move the behavior to inner_def_id.